### DISCUSSION TITLE
Why does includePath fail?

### DISCUSSION THREAD

**Discussion by xissburg** [microsoft/vscode-cpptools#7729] [Q&A]
I have all paths correctly setup in `c_cpp_properties,json` under `includePath` and when I type `#include <...` it correctly suggests the directories and files under that path but once I finish the statement with `...>` the red squiggles appear saying "cannot open source file..." which is inconsistent and shows that this is perhaps fundamentally flawed since I have always experienced this issue. What really is the problem?

**Comment by sean-mcmanus** [comment]
Is the file mentioned after "cannot open source file..." the same file that was `#include`'d or a dependency? When you run C/C++: Log Diagnostics, does the "Includes" section appear correct for the TU/source file?

**Reply by xissburg** [reply]
It complains about the specific file being included. It correctly suggests the file while typing the path after `#include` but once the statement is completed, it shows the red squiggles. Looking at the `C/C++: Log Diagnostics` I can see that the paths are missing from the `Includes` section. Though I am now noticing that it only happens for `*.hpp` files and not `*.cpp` files. Am I missing some configuration?

**Reply by sean-mcmanus** [reply]
If you're using the recursive includes feature (includePath ending with `**`) then something could be causing that to fail. The current file might have problem being parsed -- does the outline view fail to populate for that file? You may also try running C/C++: Reset IntelliSense Database in case there's some old data in there. You could also try using a non-recursive include path. This assumes you're not using compileCommands or configurationProvider for configuration.

**Reply by xissburg** [reply]
I am not using recursive include paths. Also the error is inconsistent as it shows up in some files and it doesn't in others. Resetting the database does not cause any changes.

**Reply by sean-mcmanus** [reply]
Sounds like it might be the same issue as https://github.com/microsoft/vscode-cpptools/issues/7701 ? We haven't yet figured out a repro or cause.

**Reply by xissburg** [reply]
Ok, thanks for the support!
